pkg/security/api/types.go

```diff
@@ -0,0 +1,86 @@
+package api
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+// PodSecurityPolicySubjectReview checks whether a particular user/SA tuple can create the PodSpec.
+type PodSecurityPolicySubjectReview struct {
```

Comment:
I think any user should be able to check whether a pod spec can be run in their namespace with any service account.

The primary use case here (that the API should support) is:

* I personally can check a template / pod spec is allowed when run under a DC / RC (i.e. For a service account)
* I personally can check a pod spec I'm about to create
* the deployment controller can check the pod spec and not deploy / hold / fail a deployment until the pod spec would work.

It would be better if the first two were the same API call.  The controller needs the second one.